optimum/intel/openvino/configuration.py

```diff
@@ -775,3 +775,348 @@ def to_dict(self) -> Dict[str, Any]:
+        tokenizer: Optional[str] = None,
+        processor: Optional[str] = None,
+        trust_remote_code: bool = False,
+        smooth_quant_alpha: Optional[float] = None,
+        weight_format: Optional[str] = "int8",
+        activation_format: Optional[str] = "int8",
+        **kwargs,
+    ):
+        """
+        Method for the backwards-compatible OVQuantizeOptions initialization.
+        All options are the same as those in the OVQuantizationConfig.
+        """
+        preset = "performance" if sym else "mixed"
+        advanced_parameters = {"overflow_fix": overflow_fix}
+        if smooth_quant_alpha:
+            advanced_parameters["smooth_quant_alphas"] = {"matmul": smooth_quant_alpha}
+
+        mode = None
+        if activation_format:
+            mode_map = {
+                "f8e4m3": "fp8_e4m3",
+                "f8e5m2": "fp8_e5m2",
+            }
+            mode = mode_map[activation_format]
+            preset = "performance"
+
+        return OVQuantizeOptions(
+            mode=mode,
+            preset=preset,
+            target_device="any",
+            fast_bias_correction=fast_bias_correction,
+            model_type=model_type,
+            advanced_parameters=advanced_parameters,
+        )
+
+    def to_nncf_dict(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary with the NNCF-friendly variables that are ready to use.
+        """
+        if self._nncf_dict:
+            return self._nncf_dict
+
+        if is_nncf_available():
+            mode = nncf.QuantizationMode(self.mode) if self.mode else None
+            preset = nncf.QuantizationPreset(self.preset)
+            target_device = nncf.TargetDevice(self.target_device.upper())
+            model_type = nncf.ModelType(self.model_type) if self.model_type else None
+            advanced_parameters = None
+            if self.advanced_parameters:
+                advanced_parameters = nncf.AdvancedQuantizationParameters(
+                    overflow_fix=self.advanced_parameters["overflow_fix"],
+                )
+                if "smooth_quant_alphas" in self.advanced_parameters:
+                    advanced_parameters.smooth_quant_alphas = nncf.AdvancedSmoothQuantParameters(
+                        **self.advanced_parameters["smooth_quant_alphas"]
+                    )
+
+            self._nncf_dict = {
+                "mode": mode,
+                "preset": preset,
+                "target_device": target_device,
+                "fast_bias_correction": self.fast_bias_correction,
+                "model_type": model_type,
+                "advanced_parameters": advanced_parameters,
+            }
+            return self._nncf_dict
+
+        raise ImportError("NNCF is required to execute this method. Please install nncf first.")
+
+    def to_dict(self) -> Dict:
+        return copy.deepcopy(self.__dict__)
+
+
+class OVGeneralQuantizationConfig(QuantizationConfigMixin):
+    def __init__(
+        self,
```

Comment:
Constructor of the class should be extended with `dataset`, `tokenizer` and the rest of the options.